src/HatsEligibilitiesChain.sol

```diff
@@ -0,0 +1,183 @@
+   * 40                              | hatId                     | uint256   | 32                         | HatsModule |
+   * 72                              | NUM_CONJUCTION_CLAUSES    | uint256   | 32                         | this       |
+   * 104                             | CONJUCTION_CLAUSE_LENGTHS | uint256[] | NUM_CONJUCTION_CLAUSES* 32 | this       |
+   * 104+(NUM_CONJUCTION_CLAUSES*32) | MODULES                   | address[] | NUM_MODULES * 20           | this       |
+   * ------------------------------------------------------------------------------------------------------------------+
+   */
+
+  /**
+   * @notice Get the number of conjuction clauses
+   */
+  function NUM_CONJUCTION_CLAUSES() public pure returns (uint256) {
+    return _getArgUint256(72);
+  }
+
+  /**
+   * @notice Get the a list of the lengths of every conjusction clause.
+   */
+  function CONJUCTION_CLAUSE_LENGTHS() public pure returns (uint256[] memory) {
+    return _getArgUint256Array(104, NUM_CONJUCTION_CLAUSES());
+  }
+
+  /**
+   * @notice Get all module addresses.
+   */
+  function MODULES() public pure returns (address[] memory) {
+    uint256[] memory lengths = CONJUCTION_CLAUSE_LENGTHS();
+    uint256 numClauses = lengths.length;
+    uint256 numModules;
+    for (uint256 i = 0; i < numClauses;) {
+      numModules += lengths[i];
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    address[] memory modules = new address[](numModules);
+    uint256 modulesStart = 104 + numClauses * 32;
+    for (uint256 i = 0; i < numModules;) {
+      modules[i] = _getArgAddress(modulesStart + 20 * i);
+
+      unchecked {
+        ++i;
+      }
+    }
+    return modules;
+  }
+
+  /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+  /**
+   * @notice Deploy the HatsEligibilitiesChain implementation contract and set its version
+   * @dev This is only used to deploy the implementation contract, and should not be used to deploy clones
+   */
+  constructor(string memory _version) HatsModule(_version) { }
+
+  /**
+   * @notice Get the wearer's status.
+   */
+  function getWearerStatus(address _wearer, uint256 _hatId)
+    public
+    view
+    virtual
+    override
+    returns (bool eligible, bool standing)
+  {
+    uint256 numClauses = NUM_CONJUCTION_CLAUSES();
+    uint256 clauseOffset = 104 + 32 * numClauses; // offset to current clause
+
+    bool eligibleInClause;
+    bool eligibleInModule;
+    bool standingInModule;
+    uint256 clauseIndex;
+    uint256 length;
+    address module;
+
+    while (clauseIndex < numClauses) {
+      length = _getArgUint256(104 + clauseIndex * 32); // current clause length
+
+      eligibleInClause = true;
+
+      // check eligibility and standing according to current clause
+      for (uint256 moduleIndex = 0; moduleIndex < length;) {
```

Comment:
I bet there's some efficiency to eek out by using the clauseOffset as the incrementer. That way we'd only have to increment a single variable in each iteration, not two.

Reply:
Refactored a bit in the latest commit and it did improve the gas a bit. Tried to do the same in the toggle chaining module, but it actually increased the gas so left only on the eligibilities chaining module.